src/Routes/Devices/RetryUpdatePopover.js

```diff
@@ -0,0 +1,168 @@
+import { DateFormat } from '@redhat-cloud-services/frontend-components/DateFormat';
+import apiWithToast from '../../utils/apiWithToast';
+import { updateDeviceLatestImage } from '../../api/devices';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+// import { async } from '@babel/runtime/regenerator';
+// import useIsMounted from '@data-driven-forms/common/hooks/use-is-mounted';
+
+function getDevicePopoverDescription(props) {
+  console.log(props.device);
+  if (props.device.DispatcherReason === 'The playbook failed to run.') {
+    return (
+      <div>
+        The playbook failed to run. You can retry the update or build a new one.
+        <Stack className="pf-u-mt-sm">
+          <StackItem className="pf-u-font-weight-bold">Last Seen</StackItem>
+          <StackItem> {<DateFormat date={props.lastSeen} />}</StackItem>
+        </Stack>
+      </div>
+    );
+  }
+  if (
+    props.device.DispatcherReason ===
+    'The service timed out during the last update.'
+  ) {
+    return (
+      <div>
+        The service timed out during the last update. You can retry the update
+        or build a new one.
+        <Stack className="pf-u-mt-sm">
+          <StackItem className="pf-u-font-weight-bold">Last Seen</StackItem>
+          <StackItem> {<DateFormat date={props.lastSeen} />}</StackItem>
+        </Stack>
+      </div>
+    );
+  }
+  if (props.device.DispatcherStatus === 'UNRESPONSIVE') {
+    return (
+      <div>
+        The service could not be reached via RHC. The device may communicate at
+        a later time if this is a network issues or could be an indication of a
+        larger problem.
+        <Stack className="pf-u-mt-sm">
+          <StackItem className="pf-u-font-weight-bold">Last Seen</StackItem>
+          <StackItem> {<DateFormat date={props.lastSeen} />}</StackItem>
+        </Stack>
+      </div>
+    );
+  }
+}
+
+function getDevicePopoverTitle(props) {
+  if (props.device.DispatcherReason === 'The playbook failed to run.') {
+    return <span className="pf-u-ml-xs"> Playbook error </span>;
+  }
+
+  if (
+    props.device.DispatcherReason ===
+    'The service timed out during the last update.'
+  ) {
+    return <span className="pf-u-ml-xs"> Service timed-out </span>;
+  }
+
+  if (props.device.DispatcherStatus === 'UNRESPONSIVE') {
+    return <span className="pf-u-ml-xs"> Unresponsive </span>;
+  }
+}
+
+const RetryUpdatePopover = (props) => {
+  const [isVisible, setIsVisible] = React.useState(false);
+  const dispatch = useDispatch();
+  console.log(props);
+  const isMounted = React.useRef(true);
+  React.useEffect(() => {
+    if (isMounted.current) {
+      //fetch data
+      //setData (fetch result)
+      return () => {
+        isMounted.current = false;
+      };
+    }
+  });
+  console.log(props.DeviceUUID);
+  const setUpdateModal = { setUpdateModal };
+  const statusMessages = {
+    onSuccess: {
+      title: 'Success',
```

Comment:
@mayleighnmyers heads up!

Andy is working on a mock up to align this with current update notification toast which looks like the below. So this line might require a change before this is merged.
 
---
Updating system
%device_name% as added to the queue
---

Reply:
here is the code reference @mayleighnmyers 

https://github.com/RedHatInsights/edge-frontend/blob/f3ba24ad5ffd73f4f50b3f48b490586bf37ec861/src/Routes/Devices/UpdateDeviceModal.js#L62

Reply:
We added the device name to the toast! : )